File: dota-zone/dotazone/src/main/java/br/com/dotazone/model/util/UtilActivity.java
package br.com.dotazone.model.util;

import android.app.Dialog;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.text.Html;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import br.com.dotazone.DotaZoneBrain;
import br.com.dotazone.R;
import br.com.dotazone.model.entity.Item;

public class UtilActivity {

    private static boolean isBuy;

    public static void ratingDotaZoneDialog(final Context context) {

        final SharedPreferences settings = context.getSharedPreferences(UrlUtils.PREFS_NAME, 0);
        int rating = settings.getInt(DotaZoneBrain.RATING_DOTA_ZONE, 0);

        if (rating == 3) {

            final Dialog dialog = new Dialog(context, R.style.FullHeightDialog);
            dialog.show();
            dialog.setContentView(R.layout.alert_rating_view);
            dialog.setCancelable(false);
            Button mButtonRating = (Button) dialog.findViewById(R.id.rating_start);
            Button mButtonClose = (Button) dialog.findViewById(R.id.rating_close);

            mButtonClose.setOnClickListener(new OnClickListener() {

                @Override
                public void onClick(View v) {

                    dialog.dismiss();
                }
            });
            mButtonRating.setOnClickListener(new OnClickListener() {

                @Override
                public void onClick(View v) {

                    Intent intent = new Intent(Intent.ACTION_VIEW);
                    intent.setData(Uri.parse("market://details?id=" + context.getApplicationContext().getPackageName()));
                    context.startActivity(intent);

                }
            });

        }

    }

    public static boolean showAlertBuyProVersion(final Context context) {


        final Dialog dialog = new Dialog(context, R.style.FullHeightDialog);
        dialog.show();
        dialog.setContentView(R.layout.alert_buy_view);
        dialog.setCancelable(false);
        Button buy = (Button) dialog.findViewById(R.id.buy_pro_version);
        Button close = (Button) dialog.findViewById(R.id.buy_close);

        close.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                dialog.dismiss();
            }
        });
        buy.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                isBuy = true;
                dialog.dismiss();

            }
        });

        return isBuy;


    }


    public static void showHelpBuildHero(Context context) {
        // custom dialog
        final Dialog dialog = new Dialog(context, R.style.FullHeightDialogTransparent);
        dialog.setContentView(R.layout.help_build_hero);
        RelativeLayout layoutClose = (RelativeLayout) dialog.findViewById(R.id.layout_help_build_hero);
        dialog.getWindow().setLayout(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT);
        layoutClose.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                dialog.dismiss();
            }
        });

        dialog.show();
    }

    public static void showDescriptionItem(Item item, Context context) {

        // custom dialog
        final Dialog dialog = new Dialog(context, R.style.FullHeightDialog);
        dialog.setContentView(R.layout.item_dialog_view);

        TextView itemName = (TextView) dialog.findViewById(R.id.textItemName);
        TextView itemAttribute = (TextView) dialog.findViewById(R.id.textAttribute);
        TextView itemCd = (TextView) dialog.findViewById(R.id.textCdTimer);
        TextView itemCost = (TextView) dialog.findViewById(R.id.textCost);
        TextView itemDescription = (TextView) dialog.findViewById(R.id.textDescription);
        TextView itemMana = (TextView) dialog.findViewById(R.id.textMana);
        ImageView itemIcon = (ImageView) dialog.findViewById(R.id.imageItem);
        LinearLayout itemLayoutComponents = (LinearLayout) dialog.findViewById(R.id.itemLayoutComponents);


        itemName.setText(item.getName());
        itemAttribute.setText(Html.fromHtml(item.getAttribute()));
        String cd = item.getCloudown().equals("false") ? context.getString(R.string.item_no_cd) : item.getCloudown();

        String mana = item.getMc().equals("false") ? context.getString(R.string.item_no_cd) : item.getMc();
        itemCd.setText(cd);
        itemCost.setText(item.getCost().toString());
        itemDescription.setText(Html.fromHtml(item.getDescription()));
        itemMana.setText(mana);
        int pos = item.getImageName().lastIndexOf('.');
        int idImage = context.getResources().getIdentifier(item.getImageName().substring(0, pos), "drawable",
                context.getPackageName());

        itemIcon.setImageResource(idImage);
        setImageComponents(item, itemLayoutComponents, context);

        dialog.show();

    }

    private static void setImageComponents(Item item, LinearLayout linearLayout, Context context) {

        if (!item.getComponents().isEmpty()) {

            for (String itemName : item.getComponents()) {

                int idImage = context.getResources().getIdentifier(itemName + "_lg", "drawable", context.getPackageName());
                if (idImage != 0) {
                    ImageView imageView = new ImageView(context);
                    imageView.setImageResource(idImage);
                    linearLayout.addView(imageView);
                    LinearLayout.LayoutParams param = (LinearLayout.LayoutParams) new LinearLayout.LayoutParams(UrlUtils.convertDpToPixel(50,
                            context.getResources()), UrlUtils.convertDpToPixel(32, context.getResources()));
                    imageView.setLayoutParams(param);
                }

            }
        }

    }

}

File: dota-zone/app/src/main/java/com/br/dotazone/model/util/UtilActivity.kt
package com.br.dotazone.model.util

import android.app.Dialog
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.text.Html
import android.view.View
import android.widget.*
import com.br.dotazone.DotaZoneBrain
import com.br.dotazone.R
import com.br.dotazone.model.entity.Item
import com.br.dotazone.model.util.UrlUtils.Companion.convertDpToPixel
import kotlinx.android.synthetic.main.item_dialog_view.*


object UtilActivity {
	private var isBuy = false
	fun ratingDotaZoneDialog(context: Context) {
		val settings = context.getSharedPreferences(UrlUtils.PREFS_NAME, 0)
		val rating = settings.getInt(DotaZoneBrain.RATING_DOTA_ZONE, 0)
		if (rating == 3) {
			val dialog = Dialog(context, R.style.FullHeightDialog)
			dialog.show()
			dialog.setContentView(R.layout.alert_rating_view)
			dialog.setCancelable(false)
			val mButtonRating = dialog.findViewById<View>(R.id.rating_start) as Button
			val mButtonClose = dialog.findViewById<View>(R.id.rating_close) as Button
			mButtonClose.setOnClickListener { dialog.dismiss() }
			mButtonRating.setOnClickListener {
				val intent = Intent(Intent.ACTION_VIEW)
				intent.data = Uri.parse("market://details?id=" + context.applicationContext.packageName)
				context.startActivity(intent)
			}
		}
	}

	fun showAlertBuyProVersion(context: Context?): Boolean {
		val dialog = Dialog(context!!, R.style.FullHeightDialog)
		dialog.show()
		dialog.setContentView(R.layout.alert_buy_view)
		dialog.setCancelable(false)
		val buy = dialog.findViewById<View>(R.id.buy_pro_version) as Button
		val close = dialog.findViewById<View>(R.id.buy_close) as Button
		close.setOnClickListener { dialog.dismiss() }
		buy.setOnClickListener {
			isBuy = true
			dialog.dismiss()
		}
		return isBuy
	}

	fun showHelpBuildHero(context: Context?) {
		// custom dialog
		val dialog = Dialog(context!!, R.style.FullHeightDialogTransparent)
		dialog.setContentView(R.layout.help_build_hero)
		val layoutClose = dialog.findViewById<View>(R.id.layoutHelpBuildHero) as RelativeLayout
		dialog.window!!.setLayout(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT)
		layoutClose.setOnClickListener { dialog.dismiss() }
		dialog.show()
	}

	fun showDescriptionItem(item: Item?, context: Context) {

		// custom dialog
		val dialog = Dialog(context, R.style.FullHeightDialog)
		dialog.setContentView(R.layout.item_dialog_view)
		dialog.textItemName.text = item?.name
		dialog.textAttribute.text = Html.fromHtml(item?.attribute)
		val cd = if (item?.cloudown == "false") context.getString(R.string.item_no_cd) else item?.cloudown
		val mana = if (item?.mc == "false") context.getString(R.string.item_no_cd) else item?.mc
		dialog.textCdTimer.text = cd
		dialog.textCost.text = item?.cost.toString()
		dialog.textDescription.text = Html.fromHtml(item?.description)
		dialog.textMana.text = mana
		val pos = item?.imageName?.lastIndexOf('.')?:0
		val idImage = context.resources.getIdentifier(item?.imageName?.substring(0, pos), "drawable",
				context.packageName)
		dialog.imageItem.setImageResource(idImage)
		setImageComponents(item, dialog.itemLayoutComponents, context)
		dialog.show()
	}

	private fun setImageComponents(item: Item?, linearLayout: LinearLayout, context: Context) {
		if (item != null && item.components.isNotEmpty()) {
			for (itemName in item.components) {
				val idImage = context.resources.getIdentifier(itemName + "_lg", "drawable", context.packageName)
				if (idImage != 0) {
					val imageView = ImageView(context)
					imageView.setImageResource(idImage)
					linearLayout.addView(imageView)
					val param = LinearLayout.LayoutParams(convertDpToPixel(50f,
							context.resources), convertDpToPixel(32f, context.resources))
					imageView.layoutParams = param
				}
			}
		}
	}
}
